### ISSUE TITLE
Wrong images displayed when adding images to product variants - the correct images get displayed only after second upload

### ISSUE THREAD

**Issue by danielhodis** [magento/magento2#27433]
### Preconditions (*)

1. Using Magento ver. 2.3.4
2. Working in Catalog>Products menu
3. Product set up with variants

### Steps to reproduce (*)

1. Open Product Variant from Variants list
2. Open Pictures and Videos section
3. Upload images for product variant

### Expected result (*)

1. I expect that the images displayed in Magento to be the ones I upload, i.e. images corresponding to the product I am uploading images for

### Actual result (*)

1. Instead of displaying the images I uploaded, Magento displays images from previous products I uploaded, only to display the correct images after uploading the correct images a second time

**Comment by swathishs** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @swathishs. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by engcom-Delta** [comment]
Hi @danielhodis thank you for your report, I'm not able to reproduce issue by steps you described on clean 2.4-develop
**Manual testing scenario:**
1. Navigate to Catalog / Products 
2. Select Add configurable product 
3. Select Create Configuration 
4. Select Color option and add some colors 
5. Select Apply unique images by attribute to each SKU
6. Save the product
7. Verify that all images are displayed correctly 
8. Click on the image in Current variations section and change it
9. Verify that image is saved correctly 
10. Open images and videos tab and change the image there
11. Verify that the image is changed
12. Save the product 

**Result:**
:heavy_check_mark: Correct images are displayed 
![27433](https://user-images.githubusercontent.com/51681379/98641032-c88e3800-2333-11eb-8c7a-0096abde6823.gif)

Please feel free to comment, reopen or create new ticket according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) 
if you are still facing this issue on the latest `2.4-develop` branch.
